filter/filter.go

```diff
@@ -10,20 +10,20 @@ import (
 // Filter is a special http.Handler that returns true or false depending on
 // whether subsequent handlers should continue.
 type Filter interface {
-	// ServeHTTP is like the function on http.Handler but also returns true or
-	// false depending on whether subsequent handlers should continue. If an error
-	// occurred, ServeHTTP should return the original error plus a description
-	// for logging purposes.
-	Apply(w http.ResponseWriter, req *http.Request) (ok bool, err error, errdesc string)
+	// ServeHTTP is like the function on http.Handler but also gets a context
+	// which allows it to control the progress of the filter chain.
+	Apply(w http.ResponseWriter, req *http.Request, ctx Context)
```

Comment:
I agree, Context was confusing.  I've cleaned up the syntax again.  The signature for a filter is now:

```go
Apply(w http.ResponseWriter, req *http.Request, next Next) error
```

`Next` is just a function that returns an error, so the implementor would just call `return next()`.